### ISSUE TITLE
Difference between this and another civetweb and original mongoose

### ISSUE THREAD

**Issue by agauniyal** [sunsetbrew/civetweb#63]
Asking a pretty dumb question here , but could you possibly tell the difference b/w all 3 of them.

**Comment by keithel** [comment]
TL;DR: Go to the bottom of this note and read the last two paragraphs.

The original author of Mongoose changed the license from the MIT license to a dual licensed GPLv2 or Commercial license, so if you use Mongoose, updated by the original author, you'll have to abide by either the terms of the GPLv2 in it's use (releasing any sources it's compiled with under GPLv2 or higher), or pay the author for the commercial licensing. 

Because the original sources were MIT licensed, just before it was switched to GPLv2, Github user sunsetbrew (whom's github account we're commenting on right now) forked the last MIT-licensed version of Mongoose, and continued development from there on.  Improvements from the community have been made, but it is separate from the improvements that have been made to Mongoose, so they're effectively two different products, however they share a common origin.

Sunsetbrew had other things going on in his life, so he couldn't keep up maintaining his fork. A user from the github/civetweb community - bel2125 - stepped up to maintain Civetweb under his github civetweb fork, however (I think) sunsetbrew has also been contributing since the maintenance transition to bel2125.


TL;DR:
So, if you wish to follow the original author's work, and don't mind the GPLv2 license or are ok paying for the commercial license, go ahead and use Mongoose - http://cesanta.com/libmongoose.shtml http://cesanta.com/mongoose.shtml

If you wish to follow the MIT-licensed fork (no license fee, no requirement to open source your code), then use bel2125's fork (the current official Civetweb): https://github.com/bel2125/civetweb

**Comment by wiseoldman95** [comment]
Also, I find Civetweb's interface a bit cleaner.
